activerecord/lib/active_record/core.rb

```diff
@@ -222,6 +222,56 @@ def self.connection_class_for_self # :nodoc:
       self.default_role = ActiveRecord.writing_role
       self.default_shard = :default
 
+      # Sets instances of the class to strict_loading mode. This will raise an error
+      # if a record tries to lazily load an association.
+      #
+      #   User.strict_loading! # => true
+      #   user = User.first
+      #   user.comments
+      #   => ActiveRecord::StrictLoadingViolationError
+      #
+      # ==== Parameters
+      #
+      # * +value+ - Boolean specifying whether to enable or disable strict loading.
+      # * +block+ - Proc within the block, instances of the class will set their
+      #   strict loading configuration to <tt>value</tt>. Their configuration will
+      #   be reverted to its original value outside the Proc.
+      #
+      # ==== Examples
+      #
+      #   User.strict_loading!(false) # => false
+      #   user = User.first
+      #   user.comments
+      #   => #<ActiveRecord::Associations::CollectionProxy>
+      #
+      #   User.strict_loading!(true) do
+      #     user = User.first
+      #     user.comments
+      #      => ActiveRecord::StrictLoadingViolationError
+      #   end
+      #   user = User.first
+      #   user.comments
+      #   => #<ActiveRecord::Associations::CollectionProxy>
+      #
+      #   User.strict_loading!(true) # => true
+      #   user = User.first
+      #   user.comments
+      #      => ActiveRecord::StrictLoadingViolationError
+      def self.strict_loading!(value = true, &block)
+        original_strict_loading_by_default = strict_loading_by_default
+        self.strict_loading_by_default = value
```

Comment:
As this is a class method I don't think this is threadsafe.

Reply:
That's a great point, thank you!

Is there a thread-safe mechanism to make that method viable, or do those issues make it a non-starter?

Reply:
Maybe use Threadlocal, similar to: https://github.com/rails/rails/blob/12fc98fdbb1c5038c43aa8873d961cf5046505ed/activerecord/lib/active_record/suppressor.rb#L33-L37
See discussion here: https://github.com/rails/rails/issues/18847#issuecomment-73435451

Reply:
I see searching for `IsolatedExecutionState` returns many more examples.